program/src/instruction.rs

```diff
@@ -0,0 +1,17 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::pubkey::Pubkey;
+
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug)]
+pub enum EclipseInstruction {
+    /// Accounts expected:
+    ///
+    /// 0. `[signer]`: The account of the account initialise the verification
+    /// 1. `[writable]`: Aleo transactions verification results storage account
+    /// 2. `[]`: Eclipse Program PDA account
+    /// 3. `[]`: Aleo Program account
+    /// 4. `[]`: System Program account
+    VerifyAleoTransaction {
+        tx_id: Vec<u8>,
+        aleo_program_id: Pubkey,
```

Comment:
Should the `aleo_program_id` be actually hard coded in the on-chain program? Doesn't this design introduce similar vulnerability as the Wormhole one?

I'm thinking that in this design a malicious person could submit own "aleo proof verifier" program ID w/ corresponding accounts and then fake that some given Aleo txs are verified.

Reply:
I agree with you here! good catch, I have moved the program_id into the onchain program.